controller/src/resource_controller/action.rs

```diff
@@ -151,8 +182,51 @@ async fn creation_completed_action(r: &ResourceInterface) -> Result<CreationActi
     }
 }
 
+/// Destruction for a resource may be required if the resources `DestructionPolicy` is
+/// `OnTestSuccess` or `OnTestCompletion`.
+async fn is_deletion_required(r: &ResourceInterface) -> Result<bool> {
+    let destruction_policy = r.resource().spec.destruction_policy;
+    if !matches!(
+        destruction_policy,
+        DestructionPolicy::OnTestCompletion | DestructionPolicy::OnTestSuccess
+    ) {
+        return Ok(false);
+    }
+    // If any resources still depend on this resource it should not be deleted.
+    let resources = r.resource_client().get_all().await?;
+    for resource in resources {
+        if let Some(depends_on) = resource.spec().depends_on.as_ref() {
+            if depends_on.contains(&r.name().to_string()) {
+                return Ok(false);
```

Comment:
Still just learning the code, so may be a dumb question - if this resource is marked `OnTestCompletion` or `OnTestSuccess`, but it is a dependency for another resource yet, how/where does it eventually get deleted once that dependent has been destroyed?

Reply:
Currently it doesn't until it is deleted itself. With this pr if the `DestructionPolicy` is `OnTestCompletion` or `OnTestSuccess` then it will be deleted once the resources depending on it are deleted.